### PR TITLE
feat(Video Stitcher): Add samples and tests

### PR DESCRIPTION
**Pull request by irataxy**
b:202972981

Part 1 of 3: Slates and CDN keys. See also https://cloud.google.com/video-stitcher/docs/how-to/managing-slates and https://cloud.google.com/video-stitcher/docs/how-to/managing-cdn-keys

API must be enabled in test project.

### CODE REVIEW

**Comment by amanda-tarafa** [comment]
@irataxy I've sent an internal email about the errors on this PR. Thanks!

File: media/stitcher/api/Stitcher.Samples.Tests/StitcherFixture.cs [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+    public string Hostname { get; } = "cdn.example.com";
+    public string UpdateHostname { get; } = "update.cdn.example.com";
+
+    public CdnKey TestCloudCdnKey { get; set; }
+    public string TestCloudCdnKeyId { get; set; }
+    public string CloudCdnKeyName { get; } = "gcdn-key";
+    public string CloudCdnTokenKey { get; } = "VGhpcyBpcyBhIHRlc3Qgc3RyaW5nLg==";
+    public string UpdatedCloudCdnTokenKey = "VGhpcyBpcyBhbiB1cGRhdGVkIHRlc3Qgc3RyaW5nLg==";
+    public string AkamaiTokenKey { get; } = "VGhpcyBpcyBhIHRlc3Qgc3RyaW5nLg==";
```
Comment (amanda-tarafa) [inline media/stitcher/api/Stitcher.Samples.Tests/StitcherFixture.cs:48]: Are these values that can be made public here re: security risks? Just asking, as I'm not certain what these token keys refer to.

File: media/stitcher/api/Stitcher.Samples/UpdateSlate.cs [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+public class UpdateSlateSample
+{
+    public Slate UpdateSlate(
+        string projectId, string location, string slateId, string slateUri)
+    {
+        // Create the client.
+        VideoStitcherServiceClient client = VideoStitcherServiceClient.Create();
+
+        // Build the parent location name.
+        LocationName parentLocation = new LocationName(projectId, location);
```
Comment (amanda-tarafa) [inline media/stitcher/api/Stitcher.Samples/UpdateSlate.cs:32]: You are not using this here. Check in other samples as well.
```suggestion

```

File: media/stitcher/api/Stitcher.Samples/UpdateCdnKey.cs [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+{
+    public CdnKey UpdateCdnKey(
+        string projectId, string location, string cdnKeyId, string hostname,
+        string gcdnKeyname, string gcdnPrivateKey, string akamaiTokenKey)
+    {
+        // Create the client.
+        VideoStitcherServiceClient client = VideoStitcherServiceClient.Create();
+
+        // Build the parent location name.
+        LocationName parentLocation = new LocationName(projectId, location);
```
Comment (amanda-tarafa) [inline media/stitcher/api/Stitcher.Samples/UpdateCdnKey.cs:34]: remove from here as well. I won't mark it anymore, but please, do check all samples.

File: media/stitcher/api/Stitcher.Samples/UpdateCdnKey.cs [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+using Google.Api.Gax.ResourceNames;
+using Google.Cloud.Video.Stitcher.V1;
+using Google.Protobuf;
+using Google.Protobuf.WellKnownTypes;
+
+public class UpdateCdnKeySample
+{
+    public CdnKey UpdateCdnKey(
+        string projectId, string location, string cdnKeyId, string hostname,
+        string gcdnKeyname, string gcdnPrivateKey, string akamaiTokenKey)
```
Comment (amanda-tarafa) [inline media/stitcher/api/Stitcher.Samples/UpdateCdnKey.cs:28]: nitL
```suggestion
        string gcdnKeyName, string gcdnPrivateKey, string akamaiTokenKey)

```

File: media/stitcher/api/Stitcher.Samples/UpdateCdnKey.cs [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+        LocationName parentLocation = new LocationName(projectId, location);
+
+        CdnKey cdnKey = new CdnKey
+        {
+            CdnKeyName = CdnKeyName.FromProjectLocationCdnKey(projectId, location, cdnKeyId),
+            Hostname = hostname
+        };
+
+        string path;
+        if (akamaiTokenKey != "")
```
Comment (amanda-tarafa) [inline media/stitcher/api/Stitcher.Samples/UpdateCdnKey.cs:43]: It would be more idiomatic to pass a null value than an empty string, so the check would be (and invert the code within the if-else clauses.)
```suggestion
        if (akamaiTokenKey is null)

```

File: media/stitcher/api/Stitcher.Samples.Tests/UpdateCdnKeyTests.cs [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+        private readonly UpdateCdnKeySample _updateSample;
+
+        public UpdateCdnKeyTest(StitcherFixture fixture)
+        {
+            _fixture = fixture;
+            _updateSample = new UpdateCdnKeySample();
+        }
+
+        [Fact]
+        public void UpdatesCdnKey()
```
Comment (amanda-tarafa) [inline media/stitcher/api/Stitcher.Samples.Tests/UpdateCdnKeyTests.cs:34]: Is it possible to have another test with the akamai key set to a non null value? On this same test class.

File: media/stitcher/api/Stitcher.Samples.Tests/DeleteSlateTests.cs [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+        {
+            _fixture = fixture;
+            _createSample = new CreateSlateSample();
+            _deleteSample = new DeleteSlateSample();
+            _slateId = $"{_fixture.SlateIdPrefix}-{_fixture.RandomId()}";
+
+            _createSample.CreateSlate(
+                _fixture.ProjectId, _fixture.LocationId,
+                _slateId, _fixture.TestSlateUri);
+            _fixture.SlateIds.Add(_slateId);
```
Comment (amanda-tarafa) [inline media/stitcher/api/Stitcher.Samples.Tests/DeleteSlateTests.cs:40]: nit: Add this line before actually creating the slate. It might be that creation succeeds server side, but the operation fails client side (for instance, problems handling the response), and in that case we still want to attempt to delete the slate.

File: media/stitcher/api/Stitcher.Samples.Tests/DeleteCdnKeyTests.cs [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+        {
+            _fixture = fixture;
+            _createSample = new CreateCdnKeySample();
+            _deleteSample = new DeleteCdnKeySample();
+            _cdnKeyId = $"{_fixture.CloudCdnKeyIdPrefix}-{_fixture.RandomId()}";
+
+            _createSample.CreateCdnKey(
+                _fixture.ProjectId, _fixture.LocationId,
+                _cdnKeyId, _fixture.Hostname, _fixture.CloudCdnKeyName, _fixture.CloudCdnTokenKey, "");
+            _fixture.CdnKeyIds.Add(_cdnKeyId);
```
Comment (amanda-tarafa) [inline media/stitcher/api/Stitcher.Samples.Tests/DeleteCdnKeyTests.cs:40]: Same comment as for slate

File: media/stitcher/api/Stitcher.Samples.Tests/CreateSlateTests.cs [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+            _slateId = $"{_fixture.SlateIdPrefix}-{_fixture.RandomId()}";
+        }
+
+        [Fact]
+        public void CreatesSlate()
+        {
+            // Run the sample code.
+            var result = _createSample.CreateSlate(
+                _fixture.ProjectId, _fixture.LocationId, _slateId, _fixture.TestSlateUri);
+            _fixture.SlateIds.Add(_slateId);
```
Comment (amanda-tarafa) [inline media/stitcher/api/Stitcher.Samples.Tests/CreateSlateTests.cs:42]: Same comment re adding this line before actual creation.

File: media/stitcher/api/Stitcher.Samples.Tests/CreateCdnKeyTests.cs [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+        public CreateCdnKeyTest(StitcherFixture fixture)
+        {
+            _fixture = fixture;
+            _createSample = new CreateCdnKeySample();
+            _akamaiCdnKeyId = $"{_fixture.AkamaiCdnKeyIdPrefix}-{_fixture.RandomId()}";
+            _cloudCdnKeyId = $"{_fixture.CloudCdnKeyIdPrefix}-{_fixture.RandomId()}";
+        }
+
+        [Fact]
+        public void CreatesCdnKey()
```
Comment (amanda-tarafa) [inline media/stitcher/api/Stitcher.Samples.Tests/CreateCdnKeyTests.cs:39]: Please split this test in two tests `CreateCdnKey_Akamai` and `CreateCdnKey_Google`, both tests should be on this same file.

File: media/stitcher/api/Stitcher.Samples.Tests/CreateCdnKeyTests.cs [hunk 11]
```diff
@@ -0,0 +1,58 @@
...
+        {
+            // Run the sample code and create an Akamai CDN key.
+            var result = _createSample.CreateCdnKey(
+                _fixture.ProjectId, _fixture.LocationId,
+                _akamaiCdnKeyId, _fixture.Hostname, "", "", _fixture.AkamaiTokenKey);
+            _fixture.CdnKeyIds.Add(_akamaiCdnKeyId);
```
Comment (amanda-tarafa) [inline media/stitcher/api/Stitcher.Samples.Tests/CreateCdnKeyTests.cs:45]: And same comment here about when to add the id to the collecion, and for the other key as well.

File: media/stitcher/api/Stitcher.Samples/CreateCdnKey.cs [hunk 12]
```diff
@@ -0,0 +1,71 @@
...
+        VideoStitcherServiceClient client = VideoStitcherServiceClient.Create();
+
+        // Build the parent location name.
+        LocationName parentLocation = new LocationName(projectId, location);
+
+        CdnKey cdnKey = new CdnKey
+        {
+            Hostname = hostname
+        };
+
```
Comment (amanda-tarafa) [inline media/stitcher/api/Stitcher.Samples/CreateCdnKey.cs:39]:
```suggestion

```

**Review by amanda-tarafa — CHANGES_REQUESTED** [review]
Mostly fine, just a few minor changes.

**Comment by amanda-tarafa** [comment]
@irataxy Stitcher tests are green now, but Livestream is failing. Maybe instead of deleting channels that are one day old, delete channels that are 2 hours old?

```
LiveStream.Samples.Tests.StopChannelAsyncTest.StopsChannelAsync [FAIL]

  Failed LiveStream.Samples.Tests.StopChannelAsyncTest.StopsChannelAsync [1 ms]

  Error Message:

   Grpc.Core.RpcException : Status(StatusCode="ResourceExhausted", Detail="Quota limit 'ChannelCountPerProjectPerRegion' has been exceeded. Limit: 10 in region us-central1.")
```

File: media/stitcher/api/Stitcher.Samples.Tests/StitcherFixture.cs (hunk 1 above)
Comment (irataxy) [reply media/stitcher/api/Stitcher.Samples.Tests/StitcherFixture.cs:48]: Yes, this decodes to "This is a test string."

File: media/stitcher/api/Stitcher.Samples/UpdateSlate.cs (hunk 2 above)
Comment (irataxy) [reply media/stitcher/api/Stitcher.Samples/UpdateSlate.cs]: Removed

File: media/stitcher/api/Stitcher.Samples/UpdateCdnKey.cs (hunk 4 above)
Comment (irataxy) [reply media/stitcher/api/Stitcher.Samples/UpdateCdnKey.cs]: Renamed

File: media/stitcher/api/Stitcher.Samples/UpdateCdnKey.cs (hunk 3 above)
Comment (irataxy) [reply media/stitcher/api/Stitcher.Samples/UpdateCdnKey.cs]: Removed

File: media/stitcher/api/Stitcher.Samples/UpdateCdnKey.cs (hunk 5 above)
Comment (irataxy) [reply media/stitcher/api/Stitcher.Samples/UpdateCdnKey.cs]: Done

File: media/stitcher/api/Stitcher.Samples.Tests/UpdateCdnKeyTests.cs (hunk 6 above)
Comment (irataxy) [reply media/stitcher/api/Stitcher.Samples.Tests/UpdateCdnKeyTests.cs:34]: Done

File: media/stitcher/api/Stitcher.Samples/CreateCdnKey.cs (hunk 12 above)
Comment (irataxy) [reply media/stitcher/api/Stitcher.Samples/CreateCdnKey.cs:37]: Removed

File: media/stitcher/api/Stitcher.Samples.Tests/CreateCdnKeyTests.cs (hunk 11 above)
Comment (irataxy) [reply media/stitcher/api/Stitcher.Samples.Tests/CreateCdnKeyTests.cs]: Done

File: media/stitcher/api/Stitcher.Samples.Tests/CreateCdnKeyTests.cs (hunk 10 above)
Comment (irataxy) [reply media/stitcher/api/Stitcher.Samples.Tests/CreateCdnKeyTests.cs]: Done

File: media/stitcher/api/Stitcher.Samples.Tests/DeleteCdnKeyTests.cs (hunk 8 above)
Comment (irataxy) [reply media/stitcher/api/Stitcher.Samples.Tests/DeleteCdnKeyTests.cs]: Done

File: media/stitcher/api/Stitcher.Samples.Tests/DeleteSlateTests.cs (hunk 7 above)
Comment (irataxy) [reply media/stitcher/api/Stitcher.Samples.Tests/DeleteSlateTests.cs]: Done

**Comment by irataxy** [comment]
> @irataxy Stitcher tests are green now, but Livestream is failing. Maybe instead of deleting channels that are one day old, delete channels that are 2 hours old?
> 
> ```
> LiveStream.Samples.Tests.StopChannelAsyncTest.StopsChannelAsync [FAIL]
> 
>   Failed LiveStream.Samples.Tests.StopChannelAsyncTest.StopsChannelAsync [1 ms]
> 
>   Error Message:
> 
>    Grpc.Core.RpcException : Status(StatusCode="ResourceExhausted", Detail="Quota limit 'ChannelCountPerProjectPerRegion' has been exceeded. Limit: 10 in region us-central1.")
> ```

Will do

**Review by amanda-tarafa — APPROVED** [review]
LGTM, thanks!
